tests/test_integration.py

```diff
@@ -445,225 +496,161 @@ def _get_vm():
+        expected_disk_size = self.vm_obj['disk_size_gib']
+        self.vm_obj['disk_size_gib'] += 10
 
-        expected_memory = self.vm.server_obj['memory']
-        self.vm.server_obj['memory'] += 1024
+        expected_memory = self.vm_obj['memory']
+        self.vm_obj['memory'] += 1024
 
-        self.vm.server_obj.commit()
-        self.vm.reload()
-        self.assertEqual(self.vm.server_obj['memory'], expected_memory + 1024)
+        self.vm_obj.commit()
+        self.vm_obj = self.get_vm_obj()
+        self.assertEqual(self.vm_obj['memory'], expected_memory + 1024)
 
-        vm_sync(self.vm.server_obj['hostname'])
-        self.vm.reload()
+        vm_sync(self.vm_obj['hostname'])
+        self.vm_obj = self.get_vm_obj()
 
-        self.assertEqual(self.vm.server_obj['memory'], expected_memory)
+        self.assertEqual(self.vm_obj['memory'], expected_memory)
         self.assertEqual(
-            self.vm.server_obj['disk_size_gib'],
+            self.vm_obj['disk_size_gib'],
             expected_disk_size,
         )
 
         # Shouldn't do anything, but also shouldn't fail
-        vm_sync(self.vm.server_obj['hostname'])
-        self.vm.reload()
+        vm_sync(self.vm_obj['hostname'])
+        self.vm_obj = self.get_vm_obj()
 
     def test_info(self):
-        # Not built
-        host_info(self.vm.server_obj['hostname'])
-
-        buildvm(self.vm.server_obj['hostname'])
-        host_info(self.vm.server_obj['hostname'])
-
+        host_info(self.vm_obj['hostname'])
         self.vm.shutdown()
-        host_info(self.vm.server_obj['hostname'])
-
-
-class KVMCommandTest(IGVMTest, CommandTest):
-    def setUp(self):
-        self.hv = Hypervisor(HV1)
-        self.vm = _create_vm()
-        _clean_vm([self.hv], self.vm)
+        host_info(self.vm_obj['hostname'])
 
 
 class MigrationTest(IGVMTest):
-    @classmethod
-    def setUpClass(cls):
-        cls.hv1 = Hypervisor(HV1)
-        cls.hv2 = Hypervisor(HV2)
-        cls.vm = _create_vm()
-
-        _clean_vm([cls.hv1, cls.hv2], cls.vm)
-        buildvm(cls.vm.server_obj['hostname'])
-
-    @classmethod
-    def tearDownClass(cls):
-        _clean_vm([cls.hv1, cls.hv2], cls.vm)
-
     def setUp(self):
-        # Make sure we have a clean initial state
-        self._check_vm(self.hv1, self.vm)
+        super(MigrationTest, self).setUp()
+        # Every migration gets a freshly built VM on the 1st HV
+        self.vm_obj['xen_host'] = self.hvs[0].server_obj['hostname']
+        self.vm_obj.commit()
+        buildvm(self.vm_obj['hostname'])
+        # And is performed to the 2nd HV
+        # Of course apart from migrations to automatically selected HVs
+        self.new_hv_name = HYPERVISORS[1].server_obj['hostname']
 
     def tearDown(self):
-        # Make sure we leave with a good state
-        self._check_vm(self.hv1, self.vm)
-        self._check_absent(self.hv2, self.vm)
+        super(MigrationTest, self).tearDown()
```

Comment:
If I understand super correctly removing this method would not change any behavior.

Reply:
Yes, this is the place which can be removed.